pkg/controller/service/service_controller.go

```diff
@@ -358,7 +360,14 @@ func (s *ServiceController) persistUpdate(service *v1.Service) error {
 }
 
 func (s *ServiceController) ensureLoadBalancer(service *v1.Service) (*v1.LoadBalancerStatus, error) {
-	nodes, err := s.nodeLister.ListWithPredicate(getNodeConditionPredicate())
+	var nodes []*v1.Node
+	var err error
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.TaintNodesByCondition) {
+		nodes, err = s.nodeLister.ListWithPredicate(getNodeTaintPredicate())
```

Comment:
I started to wonder about our plan of moving away from condition. Is eventual consistency (i.e. each condition will be represented by the taint as soon as the controller manages to) good enough for backward compatibility? @bgrant0607

Reply:
@gmarek @bgrant0607 - we can just let kubelet apply the taints ?

Reply:
@yastij - no, kubelet currently doesn't apply any taints, and we don't want to change that without a REALLY good reason.

@bgrant0607 - we're adding handling of Taints to ServiceController, which currently depends solely on Conditions and Unschedulable bit. Because of how we did Taints corresponding to Conditions (in NodeController, not Kubelet), there's slight delay between NodeCondition changing and Taint being applied.

Reply:
I don't have a problem with kubelet adding the taints, since they are isomorphic to conditions, but keep in mind that skew of 2 past releases is supported, so clusters may be running for at least another 6-9 months with kubelets that don't add the taints.